Patch to messaging to make the enter key insert a new-line character rather than sending the message.
Index: /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -87,7 +87,7 @@ var ChatviewAssistant = Class.create({
 		this.attachFilePicker = this.attachFilePicker.bind(this);
 		this.handleCancelAttachment = this.resetTextBox.bind(this, false);
 		this.handleTextAreaChange = this.handleTextAreaChange.bind(this);
-		this.handleTextAreaKeyUp = this.handleTextAreaKeyUp.bind(this);
+		// this.handleTextAreaKeyUp = this.handleTextAreaKeyUp.bind(this);
 		this.handleScreenStateChange = this.handleScreenStateChange.bind(this);
 		this.handleFocus = this.handleFocus.bind(this);
 		this.handleLoseFocus = this.handleLoseFocus.bind(this);
@@ -103,7 +103,7 @@ var ChatviewAssistant = Class.create({
 			hintText: $L('Enter message here...'),
 			multiline: true,
 			focus: false,
-			enterSubmits: true,
+			enterSubmits: false,
 			textFieldName: "messageTextElement",
 			className: " ",
 			changeOnKeyPress: true
@@ -146,7 +146,7 @@ var ChatviewAssistant = Class.create({
 		
 		this.cancelAttachment.observe(Mojo.Event.tap, this.handleCancelAttachment);
 	
-		this.messageTextElement.observe('keyup', this.handleTextAreaKeyUp);
+		// this.messageTextElement.observe('keyup', this.handleTextAreaKeyUp);
 		this.messageTextWidget.observe(Mojo.Event.propertyChange, this.handleTextAreaChange);
 
 		// observe keydown/keyup on the text field WIDGET to handle height changes due to text forcing it to resize
@@ -207,7 +207,7 @@ var ChatviewAssistant = Class.create({
 		this.messageTextWidget.stopObserving('keydown', this.handleTextFieldResize);
 		this.messageTextWidget.stopObserving('keyup', this.handleTextFieldResize);
 		this.messageTextWidget.stopObserving(Mojo.Event.propertyChange, this.handleTextAreaChange);
-		this.messageTextElement.stopObserving('keyup', this.handleTextAreaKeyUp);
+		// this.messageTextElement.stopObserving('keyup', this.handleTextAreaKeyUp);
 		this.Messaging.DisplayState.stopObserving(this.handleScreenStateChange);
 		this.controller.document.removeEventListener(Mojo.Event.stageActivate, this.handleFocus, false);
 		this.controller.document.removeEventListener(Mojo.Event.stageDeactivate, this.handleLoseFocus, false);
@@ -1749,12 +1749,12 @@ var ChatviewAssistant = Class.create({
 		} 
 	},
 
-	handleTextAreaKeyUp: function(event) {
+	/* handleTextAreaKeyUp: function(event) {
 		if (event && Mojo.Char.isEnterKey(event.keyCode)) {
 			this.considerForSend();
 			Event.stop(event);
 		}		
-	},
+	}, */
 
 	handleTextAreaChange: function() {
 		if (this.messageTextElement && this.messageTextElement.value == '') {
@@ -2115,4 +2115,4 @@ var ChatviewAssistant = Class.create({
 		this.controller.sceneScroller.mojo.revealBottom();
 		this.controller.sceneScroller.mojo.revealBottom();
 	}
-});
\ No newline at end of file
+});
Index: /usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -66,7 +66,7 @@ var ComposeAssistant = Class.create({
 		this.considerForSend = this.considerForSend.bind(this);
 		this.attachFilePicker = this.attachFilePicker.bind(this);
 		this.handleTextAreaChange = this.handleTextAreaChange.bind(this);
-		this.handleTextAreaKeyUp = this.handleTextAreaKeyUp.bind(this);
+		// this.handleTextAreaKeyUp = this.handleTextAreaKeyUp.bind(this);
 		this.handleCancelAttachment = this.resetTextBox.bind(this,false);		
 		this.handleRecipientsChanged = this.handleRecipientsChanged.bind(this);
 	},
@@ -99,7 +99,7 @@ var ComposeAssistant = Class.create({
 	    hintText: $L('Enter message here...'),
 	    multiline: true,
 	    focus: false,
-	    enterSubmits: true,
+	    enterSubmits: false,
 		requiresEnterKey: true,
 	    modelProperty: "value",
 		changeOnKeyPress: true
@@ -126,7 +126,7 @@ var ComposeAssistant = Class.create({
 	});
 	
 	
-	this.messageTextElement.observe('keyup', this.handleTextAreaKeyUp);
+	// this.messageTextElement.observe('keyup', this.handleTextAreaKeyUp);
 	this.messageTextWidget.observe(Mojo.Event.propertyChange, this.handleTextAreaChange);
 	
     // Observe add/delete to the addressing widget
@@ -302,12 +302,12 @@ var ComposeAssistant = Class.create({
 		} 
 	},
 
-	handleTextAreaKeyUp: function(event) {
+	/* handleTextAreaKeyUp: function(event) {
 		if (event && Mojo.Char.isEnterKey(event.keyCode)) {
 			this.considerForSend();
 			Event.stop(event);
 		}		
-	},
+	}, */
 
 	handleTextAreaChange: function() {
 		if (this.messageTextElement && this.messageTextElement.value == '') {
@@ -398,7 +398,7 @@ var ComposeAssistant = Class.create({
       that.controller.stopListening('searchList',event,that.handleRecipientsChanged);
     });
 	
-	this.messageTextElement.stopObserving('keyup', this.handleTextAreaKeyUp);
+	// this.messageTextElement.stopObserving('keyup', this.handleTextAreaKeyUp);
 	this.messageTextWidget.stopObserving(Mojo.Event.propertyChange, this.handleTextAreaChange);
 	
 	this.characterCounter.destroy();
